src/session_concurrency/payload_dataset_manager.cc

```diff
@@ -73,8 +74,8 @@ PayloadDatasetManager::GroupPayloadsBySession() const
   return session_datasets;
 }
 
-std::chrono::milliseconds
-PayloadDatasetManager::GetDelayForPayload(size_t dataset_index) const
+std::optional<std::chrono::milliseconds>
+PayloadDatasetManager::GetDelay(size_t dataset_index) const
```

Comment:
Part of this PR is fetching the delay along with other inputs (session ID, payload) _before_ the request gets sent to record for the profile export JSON.

GetDelay() previously was only called at a point in the code where it was correct to expect a delay input.

So consequently, GetDelay would throw an exception if the delay input was missing.

Now, in order to include the inputs in the profile export JSON, it was easier to have this function return an optional, to indicate whether there was a delay for a particular request.

A [different spot](https://github.com/triton-inference-server/perf_analyzer/pull/302/files#diff-364ae5c05536cf1350b7890278a35b571113c32da7ca5f465d503a78ac996031R168-R172) later in the code is used to check for the case where delay is expected and missing.